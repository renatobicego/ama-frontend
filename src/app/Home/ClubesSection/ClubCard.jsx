import Image from 'next/image'
import {Card, CardBody, CardHeader, Typography} from '@/app/utils/materialTailwind'
import Link from 'next/link'


const ClubCard = ({club}) => {
    return(
        <Link href={`/clubes/${club.nombre}`} className="mt-6" role='link'>
            <Card className="overflow-hidden h-full">
                <CardHeader 
                    className="relative w-full m-0 rounded-none"
                    floated={false}
                    shadow={false}
                >
                    <Image 
                        width={700} height={700} 
                        src={club.logoImg} 
                        alt={`Logo de club ${club.nombre}`} 
                        className="h-44 md:h-52 lg:h-64 object-contain"
                    />
                </CardHeader>
                <CardBody className="bg-hamburguer-menu-bg flex-1 p-3 md:p-6">
                    <Typography className="mb-2 text-white font-title text-base md:text-lg lg:text-xl font-medium">
                        {club.nombre}
                    </Typography>
                </CardBody>
            </Card> 
        </Link>
    )
}

export default ClubCard