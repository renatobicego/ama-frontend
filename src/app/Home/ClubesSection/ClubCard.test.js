import { cleanup, render, screen } from "@testing-library/react"
import ClubCard from "./ClubCard"
import '@testing-library/jest-dom/extend-expect'

describe('<ClubCard />', () => {

    let clubCard

    const mockClub = {
        name: "Test Club",
        imgHref: "/test/test.jpg",
        href: "/clubes/testclub/id"
    }

    beforeEach(() => {
        clubCard = render(<ClubCard club={mockClub}/>)
        
    })

    afterEach(() => {
        cleanup()
    })

    test('renders content', () => {
        expect(clubCard.container).toMatchSnapshot()

        // Render name
        const name = screen.getByText(mockClub.name)
        expect(name).toBeInTheDocument()

        // Render logo
        const logoClub = screen.getByAltText(`Logo de club ${mockClub.name}`)
        expect(logoClub).toBeInTheDocument()

        // Href is set in link component
        const linkElement = screen.getByRole('link')
        expect(linkElement.getAttribute('href')).toBe(mockClub.href)

    })
})