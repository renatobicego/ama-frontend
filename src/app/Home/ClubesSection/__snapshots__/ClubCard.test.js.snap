// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClubCard /> renders content 1`] = `
<div>
  <a
    class="mt-6"
    href="/clubes/testclub/id"
    role="link"
  >
    <div
      class="relative flex flex-col bg-clip-border rounded-xl bg-white text-gray-700 shadow-md overflow-hidden h-full"
    >
      <div
        class="bg-clip-border overflow-hidden bg-white text-gray-700 relative w-full m-0 rounded-none"
      >
        <img
          alt="Logo de club Test Club"
          class="h-44 md:h-52 lg:h-64 object-contain"
          data-nimg="1"
          decoding="async"
          height="700"
          loading="lazy"
          src="/_next/image?url=%2Ftest%2Ftest.jpg&w=1920&q=75"
          srcset="/_next/image?url=%2Ftest%2Ftest.jpg&w=750&q=75 1x, /_next/image?url=%2Ftest%2Ftest.jpg&w=1920&q=75 2x"
          style="color: transparent;"
          width="700"
        />
      </div>
      <div
        class="bg-hamburguer-menu-bg flex-1 p-3 md:p-6"
      >
        <p
          class="block antialiased mb-2 text-white font-title text-base md:text-lg lg:text-xl font-medium"
        >
          Test Club
        </p>
      </div>
    </div>
  </a>
</div>
`;
